common/chat/create.ts

```
@@ -0,0 +1,180 @@
+
+        const convertedConversation = convertTJConversation(newConversation);
+        return convertedConversation;
+    }
+
+    const convertedParticipantsConversation = convertTJConversation(participantsConversation);
+    return convertedParticipantsConversation;
+};
+
+async function ensureChatUsersExist(participants: [User, User] | User[]): Promise<void> {
+    await Promise.all(
+        participants.map(async (participant) => {
+            await getOrCreateChatUser(participant);
+        })
+    );
+}
+
+async function handleExistingConversation(
+    conversationId: string,
+    reason: ContactReason,
+    subcourseId: number | undefined,
+    conversation: TJConversation,
+    conversationInfos: ConversationInfos
+): Promise<void> {
+    if (reason === ContactReason.MATCH) {
+        await updateMatchConversation(conversationId, conversationInfos.custom.match);
+    } else if (reason === ContactReason.PARTICIPANT) {
+        await updateParticipantConversation(conversationId, subcourseId, conversation);
+    } else if (reason === ContactReason.PROSPECT) {
+        await updateProspectConversation(conversationId, subcourseId, conversation);
+    } else if (reason === ContactReason.CONTACT) {
+        await updateContactConversation(conversationId, conversationInfos);
+    }
+}
+
+async function updateMatchConversation(conversationId: string, match: any): Promise<void> {
+    const updatedConversation = {
+        id: conversationId,
+        custom: {
+            match: match,
+        },
+    };
+
+    await updateConversation(updatedConversation);
+}
+
+async function updateParticipantConversation(conversationId: string, subcourseId: number | undefined, conversation: TJConversation): Promise<void> {
+    const subcoursesFromConversation = conversation?.custom.subcourse;
+    let subcourseIds = [];
+
+    try {
+        subcourseIds = JSON.parse(subcoursesFromConversation || '[]');
+    } catch (error) {}
```

In which case would JSON.parse throw an error, and why would we silently ignore that?